### PR TITLE
feat: ✨ add syn-input

### PR DESCRIPTION
**Pull request by Gwenevere**
# Pull Request

## 📖 Description

The syn-input is a form field used to input any kind of single-line text, which could string, number etc...
This PR serves as a way to add the syn-input component with the new styling to the component library. 

### 🎫 Issues

- https://github.com/SickDesignSystem/synergy/issues/57

## 👩‍💻 Reviewer Notes

- I recommend that this is looked at by @LeaLeRose and @eileenwegner to make sure the design and functionalities are correct and work as expected.

### General

- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have made sure to follow the projects coding and contribution guides.

### Component

- [ ] I have created a new component.
- [x] I have added changes to an already existing component.
- [x] I have written integration tests (or have adjusted already existing ones).
- [x] I have validated that there are no accessibility errors.
- [x] I have created or updated any stories affected by my changes.
- [x] I have used design tokens instead of fix css values

### CODE REVIEW

**Comment by mariohamann** [comment]
@Gwenevere Please re-merge main to have a cleaner baseline. :)

**Comment by mariohamann** [comment]
@Gwenevere Please make an interactive story for the error state, so that we get a robust screenshot test.

File: packages/docs/stories/components/input.stories.ts [hunk 1]
```diff
@@ -97,6 +97,28 @@ export const ReadonlyInputs: Story = {
   },
 };
 
+/** The focus attribute provides feedback to the users, informing them that the input component is ready for use.  */
```
Comment (mariohamann) [inline packages/docs/stories/components/input.stories.ts:100]: It's not an attribute. :) Please rephrase, maybe together with @LeaLeRose so she can provide it in Figma as well.

**Comment by mariohamann** [comment]
The error state is not showing up:

CleanShot 2023-11-13 at 09 27 44

Maybe you have to wait until the input is ready, too.

**Comment by mariohamann** [comment]
@LeaLeRose @Gwenevere What I realize: "error" is not the correct term, we would have to rename it to "invalid". Could you please do in Figma and Code?

File: packages/docs/stories/components/input.stories.ts (hunk 1 above)
Comment (Gwenevere) [reply packages/docs/stories/components/input.stories.ts:100]: Yeah, I wasn't sure what to write

Comment (LeaLeRose) [reply packages/docs/stories/components/input.stories.ts:100]: the documentary is now a lot more than the things we provide in Figma or Shoelace? Is this planned? do I have to change this as well?

File: packages/components/.stylelintrc.json [hunk 2]
```diff
@@ -1,3 +1,6 @@
 {
-  "extends": "@synergy-design-system/stylelint-config-syn"
+  "extends": "@synergy-design-system/stylelint-config-syn",
+   "rules": {
```
[the remark this answers is not in the document]
Comment (schilchSICKAG) [reply packages/components/.stylelintrc.json]: This is now in `@synergy-design-system/stylelint-config-syn`. I have opted to remove the bits that where just floating around.

File: packages/components/scripts/vendorism.js [hunk 3]
```diff
@@ -155,13 +155,13 @@ const config = {
       // add custom styles to the end of `${component}.styles.ts`
       (path, content) => {
         let newContent;
-        components.forEach((component) => {
+        [...components, 'form-control'].forEach((component) => {
```
Comment (schilchSICKAG) [inline packages/components/scripts/vendorism.js:158]: Where is this coming from? What are we doing with `form-control` here?

File: packages/components/src/components/input/input.custom.styles.ts
Comment (schilchSICKAG) [inline packages/components/src/components/input/input.custom.styles.ts:1]: Currently, there is no dark-mode support as it seems?

File: packages/components/web-test-runner.config.js
Comment (schilchSICKAG) [inline packages/components/web-test-runner.config.js:1]: Why was this removed? The configuration is needed for running the integration tests.

File: packages/docs/package.json [hunk 4]
```diff
@@ -25,6 +26,8 @@
     "@storybook/web-components-vite": "^7.5.3",
     "@synergy-design-system/eslint-config-syn": "workspace:*",
     "@synergy-design-system/stylelint-config-syn": "workspace:*",
+    "@testing-library/react": "^14.1.0",
```
Comment (schilchSICKAG) [inline packages/docs/package.json:29]: What is this needed for?

File: packages/docs/src/helpers/component.ts
Comment (schilchSICKAG) [inline packages/docs/src/helpers/component.ts:1]: Would need @mariohamann opinion on this, as this currently fixes the (still persisting) problem of wrong control types. I noticed it just when using the input, even with newest storybook. I think I also know where it is coming from, but wanted a quick fix here.

Problem seems to be whitespace in `components.json` in the attributes field. It has an empty line on the definition, which may trick your plugin of thinking it is numeric (somehow?).

File: packages/components/src/components/input/input.custom.styles.ts [hunk 5]
```diff
@@ -1,5 +1,85 @@
...
+    font-size: var(--syn-font-size-2x-large);
+   }
+  
+  :host([data-user-invalid]) .input--standard {
+    border-color: var(--syn-input-border-color-focus-error);
+  }
+
+  :host([data-user-invalid]) .input--standard.input--focused:not(.input--disabled) {
+    border-color: var(--syn-input-border-color-focus-error);
+    box-shadow: 0 0 0 var(--syn-focus-ring-width) var(--syn-input-focus-ring-error);
```
Comment (schilchSICKAG) [inline packages/components/src/components/input/input.custom.styles.ts:78]: Focus-Ring-Width is currently set to `1px`, however, it should use `2px` in Figma. This is the reason it looks so strange currently: Just drawing a 1px blue border around a black border does not quite look nice ;). Please talk to @mariohamann about how to proceed here.

File: packages/components/src/styles/form-control.custom.styles.ts
Comment (schilchSICKAG) [inline packages/components/src/styles/form-control.custom.styles.ts:1]: @mariohamann: Should we add an indicator to labels (e.g. at least `cursor: pointer`?)

**Review by schilchSICKAG — CHANGES_REQUESTED** [review]

File: packages/docs/package.json (hunk 4 above)
Comment (Gwenevere) [reply packages/docs/package.json:29]: to create interactive stories, like the invalid one, where a user's action is mimicked

File: packages/assets/.env.example [hunk 6]
```diff
@@ -0,0 +1 @@
+FIGMA_PERSONAL_ACCESS_TOKEN = ""
```
[the remark this answers is not in the document]
Comment (Gwenevere) [reply packages/assets/.env.example:1]: It's not needed for the input, is needed for the fetching of assets. It is documented, yes. I just hadn't added this when I worked on the project's packages

File: packages/components/scripts/vendorism.js (hunk 3 above)
Comment (Gwenevere) [reply packages/components/scripts/vendorism.js:158]: the form-control is a component that is common to some components like the input and the textarea, since they are input components. It is a part of them.

**Review by schilchSICKAG — APPROVED** [review]
Ok from my point of view
